### PR TITLE
display DB1 when app resumes, capture response

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: src/js/constants/Timing.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(function() {
+  'use strict';
+
+  // Define global time-related constants
+  var Timing = {
+    DAYS_IN_TRIAL: 84
+  };
+
+  angular.module('medlinkApp.constants')
+         .constant('Timing', Timing);
```
Comment (michaelwehrley) [inline src/js/constants/Timing.js]: How come you chose this syntax instead of
```
.constant('Timing', {
  DAYS_IN_TRIAL: 84
});
```

Comment (ericcf) [inline src/js/constants/Timing.js]: Consistency with the other files. Each file has a named variable that corresponds to the file name.

File: src/js/services/AdherenceNotifications.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  function AdherenceNotifications($window, configurationCache,
+                                  localNotifications, Timing) {
+    this.schedule = function() {
+      var config = configurationCache.first();
+      var dateToday = $window.moment();
+      var dateStart = $window.moment(config.startDate);
+      var dayToday = dateToday.diff(dateStart, 'days') + 1;
+
+      for (var day = dayToday + 1; day <= Timing.DAYS_IN_TRIAL; day++) {
+        localNotifications.schedule({
```
Comment (michaelwehrley) [inline src/js/services/AdherenceNotifications.js]: Not sure if `angular.forEach` helps here, but I saw this the other day: https://docs.angularjs.org/api/ng/function/angular.forEach

Comment (ericcf) [inline src/js/services/AdherenceNotifications.js]: `forEach` expects an array or an object. I'd still have to build the array first.

File: test/acceptance/IosHomeSpec.js [hunk 3]
```diff
@@ -51,4 +51,14 @@ describe('ios home', function() {
       .textPresent('Settings', 'body')
         .should.eventually.be.true;
   });
+
+  it('displays the adherence dialog DB1', function() {
```
Comment (michaelwehrley) [inline test/acceptance/IosHomeSpec.js]: :smile:

File: test/unit/services/AdherenceNotificationsSpec.js [hunk 4]
```diff
@@ -3,52 +3,46 @@
...
-      adherenceNotifications.persist(record2);
-
-      var records = adherenceNotifications.fetchAll();
-      expect(records).to.have.length(2);
-      expect(records[0]).to.deep.equal(record1);
-      expect(records[1]).to.deep.equal(record2);
+  describe('#schedule', function() {
+    it('adds a notification for each subsequent day in the trial', function() {
+      adherenceNotifications.schedule();
+      expect(scheduleSpy.callCount).to.equal(2);
```
Comment (michaelwehrley) [inline test/unit/services/AdherenceNotificationsSpec.js]: Nice use of spying here.

File: test/unit/services/DialogsSpec.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  beforeEach(inject(function(_dialogs_) {
+    dialogs = _dialogs_;
+  }));
+
+  describe('#handle', function() {
+    it('clears all triggered notifications', function() {
+      dialogs.handle();
+
+      expect(clearTriggersSpy.callCount).to.equal(1);
```
Comment (michaelwehrley) [inline test/unit/services/DialogsSpec.js]: If it clears all of them, should there be a expectation that there are no notifications?  Or is testing that the method ran, sufficient?  Not sure if it is necessary to test the result of  `$window.plugin.notification.local.clearAll();`, I just didn't see/know if it should be tested.

**Comment by michaelwehrley** [comment]
Great work!

File: test/unit/services/DialogsSpec.js (hunk 5 above)
Comment (ericcf) [inline test/unit/services/DialogsSpec.js]: This is intended to be a unit test, so I'm intentionally not testing integration. Also, I'm assuming that the Cordova notification plugin works as advertised, so in my opinion it's enough to just test that it's being called as expected.

**Comment by ericcf** [comment]
@michaelwehrley Thanks for the comments!

**Pull request merged**